src/modules/task/internal/task-writer.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  CreateTaskParams,
+  DeleteTaskParams,
+  GetTaskParams,
+  Task,
+  TaskWithNameExistsError,
+} from '../types';
+import TaskRepository from './store/task-repository';
+import TaskReader from './task-reader';
+import TaskUtil from './task-util';
+
+export default class TaskWriter {
+  public static async createTask(params: CreateTaskParams): Promise<Task> {
+    const existingTask = await TaskRepository.taskDB.findOne({
+      account: params.accountId,
+      name: params.name,
+      active: true,
+    });
+    if (existingTask) {
+      throw new TaskWithNameExistsError(params.name);
+    }
+    const createdTask = await TaskRepository.taskDB.create({
+      account: params.accountId,
+      name: params.name,
+      active: true,
+    });
+    return TaskUtil.convertTaskDBToTask(createdTask);
+  }
+
+  public static async deleteTask(params: DeleteTaskParams): Promise<void> {
+    const taskParams: GetTaskParams = {
+      accountId: params.accountId,
+      taskId: params.taskId,
+    };
+    const task = await TaskReader.getTask(taskParams);
+    await TaskRepository.taskDB.findOneAndUpdate(
+      {
+        _id: task.id,
+      },
+      {
+        $set: {
+          active: false,
+        },
+      },
+      {
+        new: true,
```

Comment:
I am not clear, asking to learn.

We are specifying bunch of these values - what do they mean? What happens if we don't specify them.

Reply:
new flag just in case we are returning the updated value, 
runValidators and context flag is used just in case we add validators in the schema then on updation of document, validators don't run for updated values, so this is the way to specify that if validators will be added in schema then on updation then validations should also run on updated values also.

But I will remove it if they are not needed anymore, I just added them to use the best practice only.

Reply:
Please remove if they are absolutely not needed

Reply:
removed